primitives/core/src/traits.rs

```diff
@@ -212,3 +212,21 @@ impl CallInWasmExt {
 		Self(Box::new(inner))
 	}
 }
+
+/// Something that can spawn tasks and also can be cloned.
+pub trait CloneableSpawn: futures::task::Spawn + Send + Sync {
```

Comment:
I'm not a fan of traits that just describe the expected programatic behavior instead of some more high-level code idea.
Also this causes `Box<dyn CloneableSpawn>` to be quite widespread.

Can't we have a `struct` that will just wrap this? Like so:
```rust
#[derive(Clone)]
struct TaskExecutor(Arc<dyn futures::Task::Spawn + Send + Sync>);
impl TaskExecutor {
  pub fn spawn(...) {}
}
```
We don't even need this extra trait cause it's in `Arc` and I feel it avoids much of code repetition.

Reply:
I've considered something like that. The problem is we won't be able to have special implementation of clone in task executor this way. It might require little more than just shared reference clone in use cases I think about.